seclang/rule_parser.go

```diff
@@ -173,16 +173,19 @@ func (p *RuleParser) ParseVariables(vars string) error {
 // A operator must begin with @ (like @rx), if no operator is specified, rx
 // will be used. Everything after the operator will be used as operator argument
 func (p *RuleParser) ParseOperator(operator string) error {
+	// default operator @RX
 	if len(operator) == 0 || operator[0] != '@' && operator[0] != '!' {
-		// default operator RX
 		operator = "@rx " + operator
+	} else if operator[0] == '!' && operator[1] != '@' {
```

Comment:
I think we are missing the case ```"!"```, it would trigger an error because operator has length 1 and index [1] is out of bounds. It should be :
```if operator[0] == '!' && len(operator) > 1 && operator[1] != '@' {...```

Reply:
wow! I don't think of this case. Thank you very much!